### PR TITLE
Update ASGI docs to conform ASGI 3 specs

### PR DESCRIPTION
**Pull request by hashlash**
Note: the description and example are taken from [the ASGI docs](https://asgi.readthedocs.io/en/latest/introduction.html#how-does-asgi-work) with some modification.

### CODE REVIEW

File: docs/asgi.rst [hunk 1]
```diff
@@ -16,18 +16,16 @@ The full ASGI spec can be found at http://asgi.readthedocs.io
 Summary
 -------
 
-An ASGI application is a callable that takes a scope and returns a coroutine
-callable, that takes receive and send methods. It's usually written as a class:
+ASGI is structured as a single asynchronous callable, which takes a dict ``scope``
+and two callables ``receive`` and ``send``. In its simplest form, an application
+can be written as an asynchronous function, like this:
```
Comment (iamdevvalecha) [inline docs/asgi.rst:21]: close this!

**Review by iamdevvalecha — COMMENTED** [review]
close this

**Review by carltongibson — COMMENTED** [review]
I don't see the benefit of this change. Thanks.

**Comment by hashlash** [comment]
Shouldn't we always update the ASGI reference to the latest version?

The current docs describe the old specification of ASGI2 instead of ASGI3. ASGI2 defined as double-callable (the first one takes an argument `scope`, the later takes two arguments `send` and `receive`), while ASGI3 defined as single-callable (which takes all of the three previous arguments directly).

See django/asgiref@fae4ef0

**Comment by carltongibson** [comment]
@hashlash Yes you're right. I was looking at the text but not the code signature. Sorry!

**Review by carltongibson — APPROVED** [review]
Right, let's go with this. 

Thanks @hashlash

**Comment by hashlash** [comment]
Alright, your welcome!

**Comment by carltongibson** [comment]
Thanks again. Sorry for the brain melt. Busy week. 🤪

**Pull request merged**